frame/staking/src/benchmarking.rs

```diff
@@ -593,7 +611,7 @@ benchmarks! {
 			T::MaxNominatorRewardedPerValidator::get() as u32,
 			false,
 			true,
-			RewardDestination::Staked,
+			PayoutDestinationOpt::Split((Perbill::from_percent(50), PayoutSplitOpt::Stash)),
```

Comment:
Not sure if this matters but I would write my observation here.

We had limited the maximum nominators to be rewarded to 512 to limit i/o for `payout_stakers` call. With the split enabled, in worst case, there would be 1024 payments. Not sure how we came to the number 512 but would be good to get a second opinion about this from @kianenigma?

Also, payout_stakers_alive_staked is now benchmarked with worst case scenario of 2 payouts per nominator. This means the fees that will be charged would be a bit higher than before, even if the nominator payout type is not split. The ideal way to do this would be to benchmark this on n = `number of payouts` counting split payments as 2 instead of n = `count of nominators`. Although admittedly, may be I am being too pedantic and it might be totally okay to charge a bit higher fee (lower would be a bigger concern).

Reply:
It is relatively easy to get the number of payouts from `PayoutDestination`, something like:

```
fn num_payouts (&self) -> u32 {
		match self {
			PayoutDestination::Stake => 1,
			PayoutDestination::Deposit(_) => 1,
			PayoutDestination::Split(_) => 2,
			PayoutDestination::Forgo => 0,
		}
	}
```

But the benchmark function would need to know the total `num_payouts` value of all the `n`s  to get a precise weight - and this info is not available in the call data. I would assume this is only possible if we are only making the payout for 1 `PayoutDestination` where we could easily derive `num_payouts`.

Paged payouts will also eventually lead to a much smaller estimated fee, that will minimise this worst case weight.